.github/scripts/lint-workflow.cjs

```diff
@@ -0,0 +1,276 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+
+// This linter script checks the format of a given GitHub workflow file
+// to ensure it conforms to the format below (esp. newlines and spacing).
+// --- WORKFLOW FILE FORMAT: ---
+// # [optional - comment line(s)]]
+// name: <workflow name>
+//
+// on:
+//   ...
+//     ...
+//
+// jobs:
+//   job1_name:
+//     ...
+//     steps:
+//       - step1: ...
+//
+//       - [step2, etc.]
+//
+//   [job2_name:]
+//     ...
+//
+
+const workflowDir = path.join(__dirname, '..', 'workflows');
+const fileExts = ['.yml', '.yaml'];
+
+let ghErrOutput = '';
+let currFile = '';
+let foundError = false;
+let fileErrors = [];
+
+const logError = (line, col, msg) => {
+  // Format error message for GitHub Actions output
+  // https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message
+  const errStr =
+    `::error file=.\\github\\workflows\\${currFile},line=${line},col=${col}::${line}:${col} ${msg}`;
+  fileErrors.push(errStr);
+};
+
+const processFileErrors = () => {
+  if (fileErrors.length > 0) {
+    foundError = true;
+
+    ghErrOutput += `::group::.\\.github\\workflows\\${currFile}\n`;
+    fileErrors.forEach((err) => {
+      ghErrOutput += `${err}\n`;
+    });
+    ghErrOutput += '::endgroup::\n';
+
+    console.log(`Found ${fileErrors.length} errors in ${currFile}.`);
+  }
+  fileErrors = [];
+};
+
+const parseFile = (file) => {
```

Comment:
This is a really big function but I don't really see a sane way to break it up into subfunctions?

Reply:
> This is a really big function but I don't really see a sane way to break it up into subfunctions?

Yeah, unfortunately there's not a lot of repeated code, just a lot of logic steps.  At least it's commented? 😁